mowgli_etl/pipeline/wdc/wdc_heuristic_product_type_classifier.py

```diff
@@ -16,65 +17,66 @@ def __clean_words(line):
-                # Assume that general product name is last noun in title
-                last_noun_name = token.text
+            for token in doc:
+                if token.pos in range(92, 101):
+                    # Assume that general product name is last noun in title
+                    last_noun_name = token.text
 
-                # Assume that general product name is the first sequence of just nouns
-                if first_noun_flag == 0:
-                    if first_noun_sequence_name != "":
-                        first_noun_sequence_name += " "
-                    first_noun_sequence_name += token.text
+                    # Assume that general product name is the first sequence of just nouns
+                    if first_noun_flag == 0:
+                        if first_noun_sequence_name != "":
+                            first_noun_sequence_name += " "
+                        first_noun_sequence_name += token.text
 
-                # Assume that general product name is the last sequence of just nouns
-                if last_noun_flag == 1:
-                    last_noun_sequence_name = ""
-                    last_noun_flag = 0
-                if last_noun_sequence_name != "":
-                    last_noun_sequence_name += " "
-                last_noun_sequence_name += token.text
+                    # Assume that general product name is the last sequence of just nouns
+                    if last_noun_flag == 1:
+                        last_noun_sequence_name = ""
+                        last_noun_flag = 0
+                    if last_noun_sequence_name != "":
+                        last_noun_sequence_name += " "
+                    last_noun_sequence_name += token.text
 
-            else:
-                # Throw flag to terminate first noun sequence
-                if first_noun_sequence_name != "":
-                    first_noun_flag = 1
-                last_noun_flag = 1
+                else:
+                    # Throw flag to terminate first noun sequence
+                    if first_noun_sequence_name != "":
+                        first_noun_flag = 1
+                    last_noun_flag = 1
 
-        first_noun_sequence_name.rstrip(" ")
+            first_noun_sequence_name.rstrip(" ")
 
-        selections = [WdcProductType.option(last_noun_name, 1/3, "last_noun_heuristic"),
-                    WdcProductType.option(first_noun_sequence_name, 1/3, "first_noun_sequence_heuristic"),
-                    WdcProductType.option(last_noun_sequence_name, 1/3, "last_noun_sequence_heuristic")]
+            selections = [(last_noun_name, 1/3, "last_noun_heuristic"),
+                        (first_noun_sequence_name, 1/3, "first_noun_sequence_heuristic"),
+                        (last_noun_sequence_name, 1/3, "last_noun_sequence_heuristic")]
 
-        return WdcProductType(options=selections, source=title)
+            yield WdcProductType(options=selections, source=text, key=field)
```

Comment:
Should definitely have None instead of empty strings if you're letting it escape this class.